QuestionPaperGeneration/src/main/java/com/example/questionpapergeneration/DataBase/QuestionDao.java

```diff
@@ -0,0 +1,92 @@
+package com.example.questionpapergeneration.DataBase;
+
+import com.example.questionpapergeneration.Exceptions.BookIdAlreadyExistsExcepion;
+import com.example.questionpapergeneration.Model.Difficulty;
+import com.example.questionpapergeneration.Model.Question;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+//singleton class
+public class QuestionDao {
+
+    private HashMap<UUID, Question> questionBank = new HashMap<UUID, Question>();
+
+    private static QuestionDao INSTANCE = null;
+
+    public static QuestionDao getInstance(){
+        if(INSTANCE == null){
+            INSTANCE = new QuestionDao();
+        }
+        return INSTANCE;
+    }
+
+    public void addQuestion(Question questionPaper) {
+        if(questionBank.containsKey(questionPaper.getUid())){
+            throw new BookIdAlreadyExistsExcepion("Book Id Already Exists. Try Adding again");
+        }
+        questionBank.put(questionPaper.getUid(),questionPaper);
+        System.out.println(questionPaper.getUid()+" added to qusetionBank");
+    }
+
+    public List<Question> getQuestionOnDifficulty(HashMap<String, Integer> template_ratio) {
```

Comment:
In general you should see where to use singular or plural in the name. Here you are returning more than 1 question, so it should be plural.

Reply:
Yeah